core/src/main/java/com/amplifyframework/predictions/models/ImageEntity.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.models;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.amplifyframework.util.Immutable;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Holds the image entity detection results
+ * for the predictions category.
+ */
+public final class ImageEntity {
```

Comment:
The components of this class look like they have to do with a human / some humans / or maybe the Celebrity type?

But this class name doesn't encode that. Instead of ImageEntity, it should probably say something about people / celebrities (whatever type has age, pose, gender.)